test/timezone.lisp

```diff
@@ -69,6 +69,20 @@
          (new (timestamp+ old 24 :hour eastern-tz)))
     (is (= (* 24 60 60) (timestamp-difference new old)))))
 
+(deftest test/timezone/parse-timestring-different-timezones ()
+   (let ((*default-timezone* eastern-tz))
+     (is (timestamp=
+          (parse-timestring "2016-12-01" :offset nil)                               ;; Use default timezone
+          (parse-timestring "2016-12-01" :offset (* 60 60 -5))                      ;; Ignore timezone 
+          (parse-timestring "2016-12-01" :offset (* 60 60 -5) :timezone utc-leaps)  ;; Ignore timezone
+          (parse-timestring "2016-12-01" :offset nil :timezone eastern-tz)          ;; Use timezone
```

Comment:
This should be wrapped in a (let ((*default-timezone* +utc-zone+)) ...) because you're not actually testing here that the timezone used is different from the default.